R/zchunk_LA122.Refining_china.R

```diff
@@ -0,0 +1,197 @@
+                by = c("sector", "year")) %>%
+      # province output value = province proportion * national output value
+      mutate(value = value.x * value.y) %>%
+      select(province, sector, fuel = fuel.x, year, value) ->
+      L122.out_EJ_province_cor
+
+    # Inputs to crude oil refining - same method of portional allocations, but with multiple fuels
+    # Oil refining input fuels
+    # Calculate province oil input values
+    L122.in_EJ_R_refining_F_Yh %>%
+      filter(GCAM_region_ID == 11) %>%
+      left_join(L122.pct_province_cor, by = c("sector", "year")) %>%
+      filter(!is.na(province)) %>%
+      # province input value = province proportion * national input value
+      mutate(value = value.x * value.y, fuel = fuel.x) %>%
+      select(province, sector, fuel, year, value) ->
+      L122.in_EJ_province_cor_F
+
+    # BIOMASS LIQUIDS
+    # Downscale ethanol production to province
+
+    # Calculate the percentages of corn ethanol consumption in each province
+    biofuel_MT_province_F %>%
+      gather_years() %>%
+      filter(fuel == "ethanol") %>%
+      complete(nesting(fuel, province), year = HISTORICAL_YEARS) %>%
+      group_by(province, fuel) %>%
+      mutate(value = as.numeric(value), value = approx_fun(year, value, rule = 1)) %>%
+      ungroup() %>%
+      mutate(fuel = 'corn', sector = "corn ethanol") %>%
+      # ensure all provinces are in the data.frame even thugh they will just be NA
+      complete(sector, fuel, province = province_names_mappings$province, year = HISTORICAL_YEARS) %>%
+      group_by(year) %>%
+      mutate(value = value / sum(value, na.rm = T)) %>%
+      ungroup() %>%
+      replace_na(list(value = 0)) ->
+      L122.pct_province_btle
+
+    # Corn ethanol output by province
+    L122.pct_province_btle %>%
+      left_join_error_no_match(filter(L122.out_EJ_R_refining_F_Yh, GCAM_region_ID == 11 & sector == "corn ethanol" & fuel == "corn"),
+                by = c("sector", "year")) %>%
+      # province output value = province proportion * national output value
+      mutate(value = value.x * value.y) %>%
+      select(province, sector, fuel = fuel.x, year, value) ->
+      L122.out_EJ_province_btle
+
+    # Corn ethanol inputs by province and fuel: Repeat percentage-wise table by number of fuel inputs
+    # Corn ethanol input fuels
+    L122.in_EJ_R_refining_F_Yh %>%
+      filter(GCAM_region_ID == 11) %>%
+      left_join(L122.pct_province_btle, by = c("sector", "year")) %>%
+      filter(!is.na(province)) %>%
+      # province input value = province proportion * national input value
+      mutate(value = value.x * value.y, fuel = fuel.x) %>%
+      select(province, sector, fuel, year, value) ->
+      L122.in_EJ_province_btle_F
+
+    # Biodiesel output by province
+    # NOTE: SEDS does not cover biodiesel; using a separate EIA database for disaggregating this to provinces
```

Comment:
Replace the comment with:
# Note: CESY does not cover biodiesel, using a separate APEC dataset for disaggregating this into provinces.